### PR TITLE
Docent movement

### PR DESCRIPTION
**Pull request by Mister-Sweet**
Make docent able to move towards location after being freed + public method to set a boolean that frees the docent for movement.

### CODE REVIEW

File: Assets/Scripts/Chris/Docent.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    // Start is called before the first frame update
+    void Start()
+    {
+        rb = GetComponent<Rigidbody>();
+        destPos = wp.transform.position;
+        isFreed = true;
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (Pepijnvdliefvoort) [inline Assets/Scripts/Chris/Docent.cs]: Lege functie

Comment (Mister-Sweet) [reply Assets/Scripts/Chris/Docent.cs]: Fixed

**Review by Pepijnvdliefvoort — DISMISSED** [review]

File: Assets/Scripts/Chris/Docent.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Docent : MonoBehaviour
+{
+
```
Comment (Nosaiii) [inline Assets/Scripts/Chris/Docent.cs:5]: Witregel

File: Assets/Scripts/Chris/Docent.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public GameObject wp;
```
Comment (Nosaiii) [inline Assets/Scripts/Chris/Docent.cs]:
```suggestion
    public GameObject wayPoint;
```

File: Assets/Scripts/Chris/Docent.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    public float walkSpeed;
+
+    private Vector3 destPos;
+    private bool isFreed;
+    private Rigidbody rb;
```
Comment (Nosaiii) [inline Assets/Scripts/Chris/Docent.cs]:
```suggestion
    private Rigidbody rigidBody;
```

File: Assets/Scripts/Chris/Docent.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Docent : MonoBehaviour
+{
+
+    public GameObject wp;
+    public float walkSpeed;
+
+    private Vector3 destPos;
```
Comment (Nosaiii) [inline Assets/Scripts/Chris/Docent.cs]:
```suggestion
    private Vector3 destinationPosition;
```

File: Assets/Scripts/Chris/Docent.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        rb = GetComponent<Rigidbody>();
+        destPos = wp.transform.position;
+        isFreed = true;
+    }
+
+    private void FixedUpdate()
+    {
+        if (isFreed)
+        {
+            rb.MovePosition(Vector3.MoveTowards(transform.position, destPos, walkSpeed * Time.deltaTime));
```
Comment (Nosaiii) [inline Assets/Scripts/Chris/Docent.cs]:
```suggestion
            rb.MovePosition(Vector3.MoveTowards(transform.position, destPos, walkSpeed * Time.fixedDeltaTime));
```

**Review by Nosaiii — APPROVED** [review]

**Review by Pepijnvdliefvoort — APPROVED** [review]

**Pull request merged**